vars/artifactSetVersion.groovy

```
@@ -0,0 +1,138 @@
+
+        if(configuration.artifactType == 'appContainer' && configuration.dockerVersionSource == 'appVersion'){
+            if (script.commonPipelineEnvironment.getArtifactVersion())
+                //replace + sign if available since + is not allowed in a Docker tag
+                newVersion = script.commonPipelineEnvironment.getArtifactVersion().replace('+', '_')
+            else
+                error ("[${stepName}] No artifact version available for 'dockerVersionSource: appVersion' -> executeBuild needs to run for the application artifact first to set the artifactVersion for the application artifact.'")
+        } else {
+            def currentVersion = artifactVersioning.getVersion()
+
+            def timestamp = configuration.timestamp ? configuration.timestamp : getTimestamp(configuration.timestampTemplate)
+
+            def versioningTemplate = configuration.versioningTemplate ? configuration.versioningTemplate : configuration[configuration.buildTool].versioningTemplate
+            //defined in default configuration
+            def binding = [version: currentVersion, timestamp: timestamp, commitId: configuration.gitCommitId]
+            def templatingEngine = new SimpleTemplateEngine()
+            def template = templatingEngine.createTemplate(versioningTemplate).make(binding)
+            newVersion = template.toString()
+        }
+
+        artifactVersioning.setVersion(newVersion)
+
+        sh 'git add .'
+
+        def gitCommitId
+
+        sshagent([configuration.gitCredentialsId]) {
+            if (configuration.gitUserName  && configuration.gitUserEMail) {
+                sh "git config user.name \"${configuration.gitUserName}\""
+                sh "git config user.email \"${configuration.gitUserEMail}\""
+            }
+            try {
+                sh "git commit -m 'update version ${newVersion}'"
+            } catch (e) {
+                echo "[${stepName}]git commit failed: ${e}"
```

Should we continue here after an exception?

---

you are right. We should throw an Error. I changed this.